virttest/utils_kernel_module.py

```diff
@@ -0,0 +1,117 @@
+        :param params: Parameters to load with, 'key1=param1 key2=param2 ...'
+        """
+
+        current_config = KernelModuleHandler._load_config(self._module_name)
+        if current_config and all(x in current_config for x in params.split()):
+            logging.debug("Not reloading module. Current module configuration for %s"
+                          " already contains all requested parameters."
+                          " Requested: %s. Current: %s."
+                          % (self._module_name, params, current_config))
+            return
+
+        if not force and not self._was_loaded:
+            logging.debug("Module %s isn't loaded. Set force=True to force"
+                          " loading." % self._module_name)
+        else:
+            cmd = ""
+            if self._was_loaded:
+                cmd += 'rmmod %s; ' % self._module_name
+            cmd += 'modprobe %s %s' % (self._module_name, params)
+            logging.debug("Loading module: %s" % cmd)
+            status, output = process.getstatusoutput(cmd, shell=True,
+                                                     ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't load module %s: %s" % (
+                    self._module_name, output
+                ))
+            else:
+                self._loaded_config = params
+
+    def restore(self):
+        """Restore previous module state"""
+
+        if self._loaded_config:
+            if not self._was_loaded:
+                cmd = 'rmmod %s' % self._module_name
+            else:
+                cmd = 'rmmod %s; modprobe %s %s' % (self._module_name,
+                                                    self._module_name,
+                                                    self._config_backup)
+            logging.debug("Restoring module state: %s" % cmd)
+            status, output = process.getstatusoutput(cmd, shell=True,
+                                                     ignore_status=True)
+            if status:
+                raise exceptions.TestError("Couldn't restore module %s: %s" % (
+                    self._module_name, self._config_backup))
+
+    def _backup_config(self):
+        """
+        Check if self.module_name is loaded and read config
+
+        """
+        config = KernelModuleHandler._load_config(self._module_name)
+        if config:
+            self._config_backup = config
+            self._was_loaded = True
+        else:
+            self._was_loaded = False
+        logging.debug("Backed up %s module state (was_loaded, params)"
+                      "=(%s, %s)" % (self._module_name, self._was_loaded,
+                                     self._config_backup))
+
+    def get_was_loaded(self):
+        """ Read-only property """
+
+        return self._was_loaded
+
+    def get_config_backup(self):
+        """ Read-only property """
+
+        return self._config_backup
+
+    @staticmethod
```

Comment:
Do you have any special reason for using a static method? As far as I can see you are only calling it with `self._module_name` so I think it'd be better to just use `self._module_name` here and call it via `self`.

Also perhaps a better name would be `_parse_config` or `_serialize_params` or something like that. Anyway I am terrible at naming so feel free to ignore this second part.